actions/signed-commits/src/git/github-git/repo-tags.ts

```diff
@@ -127,3 +139,91 @@ export async function getRemoteTagNames(
 
   return tags;
 }
+
+/**
+ * Replaces the tag separator in the list of tags. Typically used to replace the @ separator
+ * with the / separator.
+ * @param tags The list of tags to update.
+ * @param separator The separator to replace the @ separator with.
+ * @returns The updated list of tags.
+ */
+export function replaceTagSeparator(
+  tags: GitTag[],
+  separator: string,
+): GitTag[] {
+  if (separator === "@") {
+    return tags;
+  }
+
+  return tags.map((tag) => {
+    const newTagName = tag.name.replace("@", separator);
+    return {
+      ...tag,
+      name: newTagName,
+      originalName: newTagName != tag.name ? tag.name : undefined,
+    };
+  });
+}
+
+/**
+ * Gets the list of major version tags from the list of tags.
+ * @param tags The list of tags to filter
+ * @param separator The separator for the list of tags
+ * @returns The list of major version tags
+ */
+export function getMajorVersionTags(
+  tags: GitTag[],
+  separator: string,
+): GitTag[] {
+  const tagNamesSeen = new Set<string>();
+  return tags.reduce((acc, tag) => {
+    const info = parseTagName(tag.name, separator);
+    if (!info) {
+      return acc;
+    }
+
+    const majorTag = `${info.pkg}${separator}v${info.major}`;
+    if (tagNamesSeen.has(majorTag)) {
+      return acc;
+    }
+    tagNamesSeen.add(majorTag);
+    acc.push({
+      name: majorTag,
+      ref: tag.ref,
+      majorVersion: true,
+    });
+    return acc;
+  }, [] as GitTag[]);
+}
+
+/**
+ * Parses a tag name into its package name and version components.
+ * @param tagName The tag name to parse
+ * @param separator The separator of the current tag
+ * @returns The parsed tag
+ */
+export function parseTagName(tagName: string, separator: string) {
+  // [a-z-]+ is the package name
+  // (\d+) is the major/minor/patch version
+  const tagRegex = new RegExp(`([a-z0-9-]+)${separator}(\\d+)\.(\\d+)\.(\\d+)`);
```

Comment:
For using this with git tags in Go "monorepos", I think we might need a `v` prefix before the semver like:

`<pkg>/v1.2.3`